code/modules/hydroponics/biogenerator.dm

```diff
@@ -132,7 +132,8 @@
 		. += mutable_appearance(icon, "[icon_state]_o_container")
 
 	if(biomass > 0)
-		var/biomass_level = min(ROUND_UP(7 * biomass / max_visual_biomass), 7)
+		var/biomass_volume_sin = sin(min(biomass/max_visual_biomass, 1) * 90)
```

Comment:
Maybe worth explaining what you're doing here with a comment, as it's not instantly obvious.